core/src/main/scala/org/apache/spark/BarrierCoordinator.scala

```diff
@@ -99,10 +101,17 @@ private[spark] class BarrierCoordinator(
     // reset when a barrier() call fails due to timeout.
     private var barrierEpoch: Int = 0
 
-    // An array of RPCCallContexts for barrier tasks that are waiting for reply of a barrier()
+    // An array of RPCCallContexts for barrier tasks that are waiting for reply of a blocking
     // call.
     private val requesters: ArrayBuffer[RpcCallContext] = new ArrayBuffer[RpcCallContext](numTasks)
 
+    // An array of allGather messages for barrier tasks that are waiting for reply of a blocking
+    // call.
+    private val allGatherMessages: ArrayBuffer[Array[Byte]] = new ArrayBuffer[Array[Byte]](numTasks)
```

Comment:
Since performance isn't a major concern here, let's make it collect an array of String type to keep the code simple.